### DISCUSSION TITLE
Downstream Analysis Question with Granger Causality

### DISCUSSION THREAD

**Discussion by zhouyi0812** [flatironinstitute/CaImAn#1208] [Q&A]
Hello!

I wonder if anyone has done downstream analysis with multivariate Granger causality? And then clustering according to the distance? 

There is a similar paper here but I am not sure if it is a good way to do that, thank you very much for your guidance!
https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-6-137

**Comment by EricThomson** [comment]
This is a big topic. It would be helpful if someone compiled, by category, the different types of downstream analyses people do with calcium data once it is extracted (and the packages they use for the analysis and find helpful). This is a rapidly growing field and the best way to analyze these huge multidimensional datasets isn't always obvious!

Getting to your questions: for clustering I'd start with scikit-learn. For granger causality stuff, I would check out the `spectral_connectivity` package and _the other packages they reference_ in their readme: https://github.com/Eden-Kramer-Lab/spectral_connectivity

I have dabbled with these measures enough to know this is a complicated topic and to treat it with respect and caution, and it is a pretty deep well. There may be newer packages out there that are really good (lots comes up when you search on this). There is some stuff in Matlab that is apparently very good that I haven't explored (from Anil Seth).

**Comment by zhouyi0812** [comment]
Hello! I just noticed and probably it is a bit late that can I still sign up for the calcium imaging data analysis camp that posted on Gitter? It will be really helpful for me.....Thank you so much!!!

**Reply by EricThomson** [reply]
Go to this site and click contact planner at the bottom: https://events.simonsfoundation.org/event/dff7cf29-7d80-461b-b780-cb8e8165bb70/summary
